torch/linalg/__init__.py

```diff
@@ -80,96 +88,97 @@
 
-.. note:: If :attr:`input` is a non-invertible matrix or non-square matrix, or batch with at least one such matrix,
-          then a RuntimeError will be thrown.
+        :func:`torch.linalg.solve` computes the action of the inverse on another matrix.
+        This is faster and more stable than computing the inverse and then multiplying
 
 Args:
-    input (Tensor): the square `(n, n)` matrix or the batch of such matrices of size
-                    `(*, n, n)` where `*` is one or more batch dimensions.
+    input (Tensor): tensor of size `(*, n, n)` where `*` is zero or more batch dimensions
+                    consisting of `(n, n)` matrices.
 
 Keyword args:
-    out (Tensor, optional): The output tensor. Ignored if ``None``. Default is ``None``.
+    out (Tensor, optional): output tensor. Ignored if `None`. Default: `None`.
+
+Raises:
+    RuntimeError: if the :attr:`input` matrix or any matrix in the batch :attr:`input` is not invertible
 
 Examples::
 
     >>> x = torch.rand(4, 4)
     >>> y = torch.linalg.inv(x)
-    >>> z = torch.mm(x, y)
+    >>> z = x @ y
     >>> z
     tensor([[ 1.0000, -0.0000, -0.0000,  0.0000],
             [ 0.0000,  1.0000,  0.0000,  0.0000],
             [ 0.0000,  0.0000,  1.0000,  0.0000],
             [ 0.0000, -0.0000, -0.0000,  1.0000]])
-    >>> torch.max(torch.abs(z - torch.eye(4))) # Max non-zero
+    >>> torch.dist(z, torch.eye(4))
     tensor(1.1921e-07)
 
     >>> # Batched inverse example
     >>> x = torch.randn(2, 3, 4, 4)
     >>> y = torch.linalg.inv(x)
-    >>> z = torch.matmul(x, y)
-    >>> torch.max(torch.abs(z - torch.eye(4).expand_as(x))) # Max non-zero
+    >>> z = x @ y
+    >>> torch.dist(z, torch.eye(4).expand_as(x))
     tensor(1.9073e-06)
 
     >>> x = torch.rand(4, 4, dtype=torch.cdouble)
     >>> y = torch.linalg.inv(x)
-    >>> z = torch.mm(x, y)
-    >>> z
-    tensor([[ 1.0000e+00+0.0000e+00j, -1.3878e-16+3.4694e-16j,
-            5.5511e-17-1.1102e-16j,  0.0000e+00-1.6653e-16j],
-            [ 5.5511e-16-1.6653e-16j,  1.0000e+00+6.9389e-17j,
-            2.2204e-16-1.1102e-16j, -2.2204e-16+1.1102e-16j],
-            [ 3.8858e-16-1.2490e-16j,  2.7756e-17+3.4694e-17j,
-            1.0000e+00+0.0000e+00j, -4.4409e-16+5.5511e-17j],
-            [ 4.4409e-16+5.5511e-16j, -3.8858e-16+1.8041e-16j,
-            2.2204e-16+0.0000e+00j,  1.0000e+00-3.4694e-16j]],
-        dtype=torch.complex128)
-    >>> torch.max(torch.abs(z - torch.eye(4, dtype=torch.cdouble))) # Max non-zero
+    >>> z = x @ y
+    >>> torch.dist(z, torch.eye(4, dtype=torch.cdouble))
     tensor(7.5107e-16, dtype=torch.float64)
+
+.. _inverse:
+    https://en.wikipedia.org/wiki/Invertible_matrix
 """)
 
 det = _add_docstr(_linalg.linalg_det, r"""
 linalg.det(input, *, out=None) -> Tensor
 
-Computes the determinant of a square matrix :attr:`input`, or of each square matrix
-in a batched :attr:`input`.
+Computes the determinant of a matrix or batch of matrices.
```

Comment:
The matrices have to be square, right?

Reply:
"determinant of a matrix or the determinants of a batch of matrices."

Reply:
Maybe the template for batched support should be:

"Computes the determinant of square matrices. Supports batched and unbatched :attr:\`input\`."

?

Reply:
I really like this template!